cheroot/test/test_server.py

```diff
@@ -439,3 +441,52 @@ def many_open_sockets(request, resource_limit):
         # Close our open resources
         for test_socket in test_sockets:
             test_socket.close()
+
+
+@pytest.mark.parametrize(
+    ('minthreads', 'maxthreads', 'error'),
+    (
+        (-1, -1, ValueError),
+        (-1, 0, ValueError),
+        (-1, 1, ValueError),
+        (-1, 2, ValueError),
+        (0, -1, ValueError),
+        (0, 0, ValueError),
+        (0, 1, ValueError),
+        (0, 2, ValueError),
+        (1, -1, None),
+        (1, 0, ValueError),
+        (1, 1, None),
+        (1, 2, None),
+        (2, -1, None),
+        (2, 0, ValueError),
+        (2, 1, ValueError),
+        (2, 2, None),
+    ),
```

Comment:
Could you also add an `ids` function for creating a nicer representation of each test factor in the pytest output?

Maybe, something like this would do:
```suggestion
    ),
    ids=lambda *params: '...'.join(params[:2]),
```

Reply:
see other comment

Reply:
It's still relevant because pytest joins parameter segments with dashes. `test_name[1...-2]` reads as a range, while `test_name[1--2]` is ambiguous/confusing.

Reply:
I had another look and I don't think this is possible. It looks like `ids` is called once per value, not once per test. As a test, I ran the tests with `ids=lambda *a, **kw: "{} {}".format(a, kw)` and it generated names like `test_threadpool_invalid_threadrange[(0,) {}-(2,) {}-('min must be > 0',) {}]`.

Reply:
Right, I keep forgetting that the `ids` behaves differently from `pytest.param(..., id=...)`. It'd make sense to address this on another level, but for now, let's drop this, so it doesn't block the merge further. It's one of those things that can be improved in many tests and could benefit from a more generic approach. This can be done separately, when somebody has inspiration for mass-refactoring...